src/main/java/picocli/CommandLine.java

```diff
@@ -6964,6 +7007,9 @@ void initDefaultValueProvider(Class<? extends IDefaultValueProvider> value, IFac
             void initExitCodeOnInvalidInput(int exitCode)       { if (initializable(exitCodeOnInvalidInput, exitCode, ExitCode.USAGE)) { exitCodeOnInvalidInput = exitCode; } }
             void initExitCodeOnExecutionException(int exitCode) { if (initializable(exitCodeOnExecutionException, exitCode, ExitCode.SOFTWARE)) { exitCodeOnExecutionException = exitCode; } }
             void updateName(String value)                   { if (isNonDefault(value, DEFAULT_COMMAND_NAME))                {name = value;} }
+            void updateModelTransformer(Class<? extends IModelTransformer> value) {
+                if (isNonDefault(value, NoOpModelTransforemer.class)) { this.modelTransformer = DefaultFactory.create(defaultFactory(), value); }
```

Comment:
Please use the `factory` that is passed as parameter to the `updateCommandAttributes` method, not the default factory.

Reply:
Ok.